app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java

```diff
@@ -1,18 +1,200 @@
+import android.preference.PreferenceActivity;
+import android.preference.PreferenceGroup;
+import android.preference.PreferenceManager;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.app.ActionBar;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatDelegate;
+import android.support.v7.widget.Toolbar;
 import android.view.Menu;
+import android.view.MenuInflater;
 import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.DatePicker;
+import android.widget.TimePicker;
+import android.widget.Toast;
+import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
+
+
+import net.danlew.android.joda.JodaTimeAndroid;
+
+import org.joda.time.DateTime;
 
 import uy.com.bix.app.smsproject.R;
+import uy.com.bix.app.smsproject.classes.AlertReceiver;
 
-public class SettingsActivity extends AppCompatActivity {
+public class SettingsActivity extends PreferenceActivity implements OnSharedPreferenceChangeListener {
+
+	int expirationYear, expirationMonth, expirationDay, expirationHour, expirationMinute;
+	boolean isLastDay, isActive;
+	private AppCompatDelegate mDelegate;
+
+	static final int DATE_DIALOG_ID = 0;
+	static final int TIME_DIALOG_ID = 1111;
+	static final String MESSAGE_KEY = "Message";
+	static final String NUMBER_KEY = "Phone";
+	static final String MAX_KEY = "Max";
+	static final String DATE_KEY = "btnDateFilter";
+	static final String TIME_KEY = "btnTimeFilter";
+	public static Context contextOfApplication;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
+		getDelegate().installViewFactory();
+		getDelegate().onCreate(savedInstanceState);
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.activity_settings);
+		addPreferencesFromResource(R.xml.settings_activity);
+		contextOfApplication = getApplicationContext();
+
+		Preference btnDateFilter = findPreference("btnDateFilter");
+		btnDateFilter.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+			@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+			@Override
+			public boolean onPreferenceClick(Preference preference) {
+				showDialog(DATE_DIALOG_ID);
+				return false;
+			}
+		});
+
+		Preference btnTimeFilter = findPreference("btnTimeFilter");
+		btnTimeFilter.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+			@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+			@Override
+			public boolean onPreferenceClick(Preference preference) {
+				showDialog(TIME_DIALOG_ID);
+				return false;
+			}
+		});
+
+		JodaTimeAndroid.init(this);
+
+		// Get saved user settings
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
+
+		// The month in the date picker is in 0-11 range
+		expirationMonth = settings.getInt("Month", 1) - 1;
+		expirationDay = settings.getInt("Day", 1);
+		expirationHour = settings.getInt("Hour", 23);
```

Comment:
**In the future ** will be great to have constants for this kind of values.

For example:

`private static int DEFAULT_HOUR = 23;`

Or better in a configuration file.